AdminConsole Home

israelCyabukombe opened the pull request:
Created a better detailed view of the AdminConsole page. 

Modified files index.cshtml add a font attribute to site.css and added another to the bootrasp- 3.3.7.js file. These were minor changes

dubeme committed:
Add index page

Using this file to test the new requirement for requesting code review
via pull request.

dubeme committed:
Add lodash.js

Using this for it's vast selection of utility functions

quazi-irfan committed:
Merge pull request #59 from dubeme/development

Add index page

israelCyabukombe committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

israelCyabukombe committed:
Successful check-in, (Waiver signed, but account suspended)  #55

This is addressing issue #55

The page will provide an output message that says:-
"Check-in Successfull"
"This Account has been suspended! "

israelCyabukombe committed:
Merge branch 'master' into development

dubeme committed:
Merge pull request #60 from israelCyabukombe/development

Successful check-in, (Waiver signed, but account suspended)

sakurano92 committed:
UI for login screen #19

UI for login page, currently working progress

basic work on it

will be added alot. (2016/11/12)

sakurano92 committed:
testing github #19

Pull request update

added else if statement to see employee(2016/11/12)

israelCyabukombe committed:
Successful check-in, (Haven't signed in, in over 3 months (90 days)) #56

issue #56

This will output a page:
"Check-in Succsful"
"The user hasn't signed in for over 90 Days!!!"

quazi-irfan committed:
Merge pull request #61 from sakurano92/development

UI for login screen #19

AboodV2 committed:
Successful check-in, (Waiver signed, but certification expired)

AboodV2 committed:
removed unused file

israelCyabukombe committed:
fixed from quazi's review #56

israelCyabukombe committed:
Merge pull request #62 from israelCyabukombe/development

Successful check-in, (Haven't signed in, in over 3 months (90 days))

dubeme committed:
Merge remote-tracking branch 'refs/remotes/origin/development' into sleepy-hacks/development

dubeme committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

dubeme committed:
Merge branch 'sleepy-hacks/development' into development

dubeme committed:
Merge remote-tracking branch 'upstream/development' into development

dubeme committed:
Add git ignore

Start off with visual studio template from
(https://www.gitignore.io/api/visualstudio)

dubeme committed:
Add visual studio solution

dubeme committed:
Create MVC6 web app

dubeme committed:
Remove default frontend dependenciew

dubeme committed:
Sort setting (vs is complaining about it)

dubeme committed:
Manually add frontend dependenciew

* bootstrap
* lodash
* jQuery

AboodV2 committed:
improved version

dubeme committed:
Remove Home controller

Just cleaning up template code

dubeme committed:
Add new features

* Add new controllers
* Fix indentations

dubeme committed:
Update route mapping

dubeme committed:
Remove default JS file

dubeme committed:
Remove some generated layout code

dubeme committed:
Add check-in view

dubeme committed:
Add new class library project

This will be used to abstract most business logic

dubeme committed:
Remove site.js from minifier

dubeme committed:
Add some JS file

dubeme committed:
Refactor namespace for web app

dubeme committed:
Refactor namespace for wama.core

dubeme committed:
Add entity framework dependency

dubeme committed:
Add  some POCOs  and Service classes

dubeme committed:
Add lodash.js (#64)

Using this for it's vast selection of utility functions

AboodV2 committed:
Successful check-in, (Waiver signed, but certification expired) (#63)

* Successful check-in, (Waiver signed, but certification expired)

* removed unused file

dubeme committed:
Fixx minor issues

* Update namespace references
* Refactor parameter name
* Update layout

dubeme committed:
Resolve bug with logging DLL not found

dubeme committed:
Add EF command line tools

dubeme committed:
Register services for dependency injection

dubeme committed:
Refactor Namespace to better match usage of model

Since these models will be used to map db tables, it makes I'm turn between naming it's namespace.
I contemplated between ViewModel, POCO, & DTO. DTO seems to make more sense at the moment.

Might change later

dubeme committed:
Add base to represent a database table row

sakurano92 committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

sakurano92 committed:
Merge remote-tracking branch 'upstream/development' into development

sakurano92 committed:
few changed on UI

moved directory to front end
1. Login for test purpose created
2. allow enter along side with click to login
3. pw authentication created
4. background css created < use this for background use

sakurano92 committed:
few changed on UI

moved directory to front end
1. Login for test purpose created
2. allow enter along side with click to login
3. pw authentication created
4. background css created < use this for background use

dubeme committed:
Revert "changed for UI and added" (#66)

sakurano92 committed:
some how old code was in my local...

This is REAL updated one... sorry

sakurano92 committed:
background css is applied

background css is applied for login and authentication

sakurano92 committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

sakurano92 committed:
waiver adaptation#19

minor change to adapt to waiver

sakurano92 committed:
background

background is inserted

sakurano92 committed:
Sign waiver page#51

simple sign waiver page

sakurano92 committed:
checkin success

basic checkin success page if no problem.

dubeme committed:
Enable entity framework migration

dubeme committed:
Refactor namespace to match coding style

dubeme committed:
Add viewmodels for user account

dubeme committed:
Add resource file for managing strings

dubeme committed:
Just another refactor

* Add abstract method to translate a user account viewmodel to it's
   appropriate DTO
* Update namespace

dubeme committed:
Just another refactor

* Add extension methods for user viewmodel
* Move converter method to extension method

dubeme committed:
Update user account service to use view model

dubeme committed:
Configure DB context with proper connection string

AboodV2 committed:
Merge pull request #67 from sakurano92/development

Some how old login UI html was in there...

quazi-irfan committed:
Add Check In Success with Waiver signed #54

quazi-irfan committed:
Add text to the title #54

quazi-irfan committed:
Add Check In Success with Waiver not signed #58

dubeme committed:
Add view/action for creating a patron

dubeme committed:
Refactor, remove whitespaces and unused namespaces

dubeme committed:
Add DTO's for certification and waiver

dubeme committed:
Refactor, rename class

dubeme committed:
Update property name

dubeme committed:
Update property name

dubeme committed:
Add more DB set to db context

dubeme committed:
Add provider for db context

dubeme committed:
Add more extension methods

dubeme committed:
Add new services with some method signature update

dubeme committed:
Introduce virtual navigation propreties

More information, http://stackoverflow.com/a/8542879

dubeme committed:
Configure relationships within entities via fluent API

dubeme committed:
Update EF migration

dubeme committed:
Add string to use as key for sending error messages to views

dubeme committed:
Add more view models

dubeme committed:
Add dependency resolover

dubeme committed:
Refactor view code; introduce partial view for displaying errors

dubeme committed:
Refactor, update property name

dubeme committed:
Add more actions

dubeme committed:
Update logic for creating user

dubeme committed:
Merge branch 'development' of https://github.com/dubeme/WAMA into development

dubeme committed:
Add lodash.js (#69)

Using this for it's vast selection of utility functions

dubeme committed:
Introduce the MVC 6 app (#70)

* Add lodash.js

Using this for it's vast selection of utility functions

* Add git ignore

Start off with visual studio template from
(https://www.gitignore.io/api/visualstudio)

* Add visual studio solution

* Create MVC6 web app

* Remove default frontend dependenciew

* Sort setting (vs is complaining about it)

* Manually add frontend dependenciew

* bootstrap
* lodash
* jQuery

* Remove Home controller

Just cleaning up template code

* Add new features

* Add new controllers
* Fix indentations

* Update route mapping

* Remove default JS file

* Remove some generated layout code

* Add check-in view

* Add new class library project

This will be used to abstract most business logic

* Remove site.js from minifier

* Add some JS file

* Refactor namespace for web app

* Refactor namespace for wama.core

* Add entity framework dependency

* Add  some POCOs  and Service classes

* Fixx minor issues

* Update namespace references
* Refactor parameter name
* Update layout

* Resolve bug with logging DLL not found

* Add EF command line tools

* Register services for dependency injection

* Refactor Namespace to better match usage of model

Since these models will be used to map db tables, it makes I'm turn between naming it's namespace.
I contemplated between ViewModel, POCO, & DTO. DTO seems to make more sense at the moment.

Might change later

* Add base to represent a database table row

* Enable entity framework migration

* Refactor namespace to match coding style

* Add viewmodels for user account

* Add resource file for managing strings

* Just another refactor

* Add abstract method to translate a user account viewmodel to it's
   appropriate DTO
* Update namespace

* Just another refactor

* Add extension methods for user viewmodel
* Move converter method to extension method

* Update user account service to use view model

* Configure DB context with proper connection string

* Add view/action for creating a patron

* Refactor, remove whitespaces and unused namespaces

* Add DTO's for certification and waiver

* Refactor, rename class

* Update property name

* Update property name

* Add more DB set to db context

* Add provider for db context

* Add more extension methods

* Add new services with some method signature update

* Introduce virtual navigation propreties

More information, http://stackoverflow.com/a/8542879

* Configure relationships within entities via fluent API

* Update EF migration

* Add string to use as key for sending error messages to views

* Add more view models

* Add dependency resolover

* Refactor view code; introduce partial view for displaying errors

* Refactor, update property name

* Add more actions

* Update logic for creating user

dubeme committed:
Remove old front end project

dubeme committed:
Fix incorrect account type bug

dubeme committed:
Register some dependency resolvers

dubeme committed:
Add method to create login credential

dubeme committed:
Add ability to create login credential after creating an account

dubeme committed:
Just another refactor, update login action logic

dubeme committed:
Sync up with sleep-hacks development

Merge remote-tracking branch 'sleepy-hacks/development' into development

dubeme committed:
Merge pull request #74 from dubeme/development (#71, #72)

Added new feature(s), and also removed old work

dubeme committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

dubeme committed:
Add unit test project

dubeme committed:
Update default namespace

Used this to get this test project setup
(http://stackoverflow.com/questions/37012757/attach-unit-tests-to-asp-net-core-project)

dubeme committed:
Update test project

The initial project wasn't running the tests. Hence, I dug around and
decided to go with the xunit testing framework. It also looks like it's
the way that the .net team at microsoft is going.

https://xunit.github.io/

dubeme committed:
Add tests for some viewmodel

dubeme committed:
Add unit test project with some tests (#78)

* Add unit test project

* Update default namespace

Used this to get this test project setup
(http://stackoverflow.com/questions/37012757/attach-unit-tests-to-asp-net-core-project)

* Update test project

The initial project wasn't running the tests. Hence, I dug around and
decided to go with the xunit testing framework. It also looks like it's
the way that the .net team at microsoft is going.

https://xunit.github.io/

* Add tests for some viewmodel

dubeme committed:
Revert "Add some unit tests"

dubeme committed:
Merge pull request #79 from sleepy-hacks/revert-78-development

Revert "Add some unit tests"

dubeme committed:
Merge pull request #80 from dubeme/development

Create unit test project, also add some tests

dubeme committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

dubeme committed:
Add test for CheckinService

dubeme committed:
Merge pull request #81 from dubeme/development

Add unit test for CheckInService

dubeme committed:
Introduce new base controller class

This class should help us share more code across our controllers.

dubeme committed:
Setup admin console

dubeme committed:
Add admin console tools

* User tool
* Inventory tool
* Report tool
* Clinic tool

sakurano92 committed:
artificial delay before redirect to checkin #86

simple timer added need to work on but will delay for 10 seconds for now

dubeme committed:
Add property to track if logged in

THis field is of no use at the moment. Since when some checks in,
they don't check out when they're done. Hence, this might just be handy
for employees. You can see who is on clock.

dubeme committed:
Add more helper methods to service classes

dubeme committed:
Refactor logic for checking is member ID has a login credential

dubeme committed:
Create extension method class for strings

dubeme committed:
Switch to proper base class

dubeme committed:
Add strings to assit user tool

dubeme committed:
Add logic to retrieve appropriate user accounts

dubeme committed:
Add views to assit with displaying user accounts

dubeme committed:
Update view code to properly display user account

dubeme committed:
Update UI

dubeme committed:
Update database to reflect changes in DTOs

sakurano92 committed:
Delay added after patron created #86

Needs more work on

quazi-irfan committed:
Add login ID validation in Checking page

quazi-irfan committed:
Add first and lsat name validation in Create Patron page

sakurano92 committed:
#88 User Id entered to create page

dubeme committed:
Add css to format user account data

sakurano92 committed:
#88 minor fix

minor fix that dubeme ask is done

dubeme committed:
Merge pull request #91 from sakurano92/development

Fix issues #86 and #88

dubeme committed:
Update error message style

dubeme committed:
Add template for displaying user account details

dubeme committed:
Add user account Edit/View views

dubeme committed:
Refactor, remove duplicate code

dubeme committed:
Use async version for getting user account

dubeme committed:
Add method for setting active console tool

dubeme committed:
Add glyphs next to the tools

quazi-irfan committed:
Add unicode support for first and last name in Patron Check in page

dubeme committed:
Refactor, remove dead code

dubeme committed:
Update controller logic to simplfy view code

dubeme committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

israelCyabukombe committed:
Add Gender field to the Create Account page

israelCyabukombe committed:
Add the Gender to the UserAccountViewModel Class

AboodV2 committed:
Login error message if the account does not exit

AboodV2 committed:
Merge branch 'development' of https://github.com/sleepy-hacks/WAMA.git

Conflicts:
	app/wama.web/Controllers/CheckInController.cs

AboodV2 committed:
Merge remote-tracking branch 'refs/remotes/origin/development' into sleepy-hacks/development

israelCyabukombe committed:
Add the Certification field to Create.cs

israelCyabukombe committed:
Add the Certification to UserAccountViewModel.cs

dubeme committed:
Merge pull request #96 from israelCyabukombe/development

Add more fields for create account #85

dubeme committed:
Merge pull request #92 from iamcreasy/development

Add login ID validation in Checking page #87

dubeme committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

dubeme committed:
Merge pull request #97 from dubeme/development

Create layout for admin tools #90

dubeme committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

quazi-irfan committed:
Revert last and first name regex to accepting English letter only at Create Patron Page

AboodV2 committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

dubeme committed:
Update to async version (issue #100)

AboodV2 committed:
update DB connection string

AboodV2 committed:
Adding Error message when the ID does not exit

AboodV2 committed:
revert changes to core files

dubeme committed:
Merge pull request #101 from AboodV2/development

Add error the login page when the account does not exit #83

dubeme committed:
Update logic for delaying account creation

dubeme committed:
Fix null reference bug (user doesnt exist)

dubeme committed:
Persist member ID on error while creating account #102

dubeme committed:
Display error while creating account with existing member ID #103

dubeme committed:
Merge pull request #99 from iamcreasy/development

Revert last and first name regex to accepting English letter only at …

dubeme committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

dubeme committed:
Syncing with sleepy-hacks (#104)

dubeme committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

dubeme committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

dubeme committed:
Bug fixes and small features (#100, #102, #103) (#105)

Bug fixes and small features (#100, #102, #103, #106)

dubeme committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

dubeme committed:
Remove frontend folder (AGAIN!!!)

dubeme committed:
Remove frontend folder (AGAIN!!!) (#107)

* Update to async version (issue #100)

* Update logic for delaying account creation

* Fix null reference bug (user doesnt exist)

* Persist member ID on error while creating account #102

* Display error while creating account with existing member ID #103

* Remove frontend folder (AGAIN!!!)

dubeme committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

dubeme committed:
Remove frontend folder (AGAIN!!!) 2nd try (#108)

* Update to async version (issue #100)

* Update logic for delaying account creation

* Fix null reference bug (user doesnt exist)

* Persist member ID on error while creating account #102

* Display error while creating account with existing member ID #103

* Remove frontend folder (AGAIN!!!)

dubeme committed:
Add more viewmodels and extension methods

dubeme committed:
Update DTOs as per new requirements

dubeme committed:
Update ViewModels as per new requirements

dubeme committed:
Update service classes as per new requirements

dubeme committed:
Just another refactor

Update more logic, especially in the extension methods. Also clean up
the services

dubeme committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

dubeme committed:
Update DTOs as per new features (#109) (#110)

* Update to async version (issue #100)

* Update logic for delaying account creation

* Fix null reference bug (user doesnt exist)

* Persist member ID on error while creating account #102

* Display error while creating account with existing member ID #103

* Remove frontend folder (AGAIN!!!)

* Add more viewmodels and extension methods

* Update DTOs as per new requirements

* Update ViewModels as per new requirements

* Update service classes as per new requirements

* Just another refactor

Update more logic, especially in the extension methods. Also clean up
the services

dubeme committed:
Revert "Update DTOs as per new features (#109)"

dubeme committed:
Merge pull request #111 from sleepy-hacks/revert-110-development

Revert "Update DTOs as per new features (#109)"

dubeme committed:
Merge pull request #112 from dubeme/development

Update DTOs as per new features (#109) - redo

dubeme committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

dubeme committed:
Update Foreign keys, and also table fields

dubeme committed:
Merge pull request #122 from dubeme/development

Fix DB bugs (#121)

dubeme committed:
Update to .NET 1.1.0

dubeme committed:
Add utility classes for generating CSV from an object

dubeme committed:
Perfrom a join to retrieve member when getting check-in activity

dubeme committed:
Add new features

* Generate check-in reports
* Download check-in reports as CSV

sakurano92 committed:
#126 create user url after 1st attempt

I still need to center it but dont know how... need help on centering
those link

sakurano92 committed:
Merge branch 'development' into development

sakurano92 committed:
#126 create link after 1st attempt

fixed as requested!

dubeme committed:
Fix bug with missing fields

dubeme committed:
Merge pull request #131 from dubeme/development

Fix bugs with missing fields

sakurano92 committed:
Merge remote-tracking branch 'sleepy-hacks/development' into development

sakurano92 committed:
#126 change on 1st attempt

changes on 1st attempt checkin
changed the name convention

sakurano92 committed:
#126 naming convention changed

changed naming convention by request

dubeme committed:
Merge pull request #128 from sakurano92/development

#126 create user url after 1st attempt

dubeme committed:
Merge remote-tracking branch 'refs/remotes/sleepy-hacks/development' into development

dubeme committed:
Fix bug with Violation of DB table field contract (#135)

dubeme committed:
Update patron account creation page (#136)

* Update DB schema
* Add new fields with updated attributes
* Append * in required fields using CSS
* Switch to using enum for Gender and Institution
* Fix bugs in account creation

dubeme committed:
Fix bug for when member ID is missing in URL

dubeme committed:
Remove dead code

dubeme committed:
Set default dates for the filter

dubeme committed:
Update Listserv tool, to display the appropriate data

dubeme committed:
Merge pull request #138 from dubeme/development

Bug fixes

dubeme committed:
Set check-in as default report tool

dubeme committed:
Remove unused report tools

sakurano92 committed:
#139 Delay creating for non-approved user

If user are not approved by admin or employee will not be checked in.

sakurano92 committed:
#140 bug fix for creating user

Create user bug is fixed.
bug was: didnt check for 7 digit long when creating user
even user entered id less or more then 7 digit
it still generated in datatabase.

sakurano92 committed:
pull for avoid murge conflict

quazi-irfan committed:
Check if account is approved

dubeme committed:
Merge pull request #141 from sakurano92/development

#139 Delay creating for non-approved user

dubeme committed:
Merge pull request #146 from iamcreasy/account_approval

Check if account is approved

AboodV2 committed:
Adding Pending/Suspended User Implementation To UserTool

AboodV2 committed:
Add New User to UserTool

AboodV2 committed:
Merge branch 'upstream/development' into sleepy-hacks/development

AboodV2 committed:
Merge remote-tracking branch 'refs/remotes/origin/development' into upstream/development

AboodV2 committed:
bug fix for Suspended/Pending Users

AboodV2 committed:
The request change from the old pull request

dubeme committed:
Switch to using interface for generating CSV

I choose this method to give the developer the power to properly
customize the properties of a POCO that will be serialized into a CSV.

dubeme committed:
Add API for getting aggregate check-in data

dubeme committed:
Stop join on member property

dubeme committed:
Update report tool

* Add ability to get aggregated report
* Refactor common code
* Improve download experience

dubeme committed:
Six minor bugs

* Refactor property name
* Update strings

dubeme committed:
Merge pull request #151 from dubeme/development

Update report tool and bug fixes

dubeme reviewed (changes requested):
This is too big for a pull request.

I'll suggest, create a brand new fork of this project to your GitHub profile. Then copy over the new changes you made into the new copy you've got. Then send your pull request.

dubeme committed:
Merge pull request #150 from AboodV2/development

requested changes from the old pull request

dubeme committed:
Reformat with CodeMaid

dubeme committed:
Append date to download file

dubeme committed:
Add pagination on user-tool table-data

dubeme committed:
Update logic for generating tabular view for user accounts

dubeme committed:
Remove duplicate code

This 2 files is exactly the same as "UserAccountTableDisplay". I feel
these were created since wee needed to include a button for adding a
new user.

This behaviour can be achieved in another way.

dubeme committed:
Introduce partial view to render top bar on user tool

dubeme committed:
Update user account view page

dubeme committed:
Merge pull request #153 from dubeme/development

Code clean up and minor bug fixes
